File: C++/p_1_fibonacci.cpp
#include <iostream>
using namespace  std;
int count = 2;
int fibonacci(int n);
// 2nd way
/*
void fibonacci(int prev1, int prev2) {
    if (count <= 19) {
        int newFib = prev1 + prev2;

        cout << newFib << " ";

        prev2 = prev1;
        prev1 = newFib;
        count ++;

        fibonacci(prev1, prev2);
    } else {
        return ;
    }
}
*/
int main(){
    // cout << 0 << " " << 1 << " ";
    // fibonacci(1, 0);

    /*
    //                    1st way

    int prev1 = 1, prev2 = 0;
    cout << prev2  << " " << prev1 <<" ";

    for (int i = 0; i<18; i++) {
        int newFib = prev1 + prev2;
        cout << newFib << " ";
        prev2 = prev1;
        prev1 = newFib;
    }
    */

    //              3rd way
    cout << fibonacci(19);
    
    return 0;
}
int fibonacci(int n) {
    if (n<=1) {
        return n;
    } else {
        return fibonacci(n-1) + fibonacci(n-2);
    }
}

File: Python/p_1_fibonacci.py
#                     first way 
'''fib2 = 0
fib1 = 1
print(fib2, fib1, end=" ")
for i in range(18):
    newFib = fib1 + fib2
    print(newFib, end=" ")
    fib2 = fib1
    fib1 = newFib
'''
#                      second way 

'''
print(0, 1, end=" ")
count = 2

def fibonacci(prev1, prev2):
    global count
    if count<=19:
        newFib = prev1 + prev2
        print(newFib, end=" ")
        prev2 = prev1
        prev1 = newFib
        count +=1
        
        fibonacci(prev1, prev2)
    else:
        return 

fibonacci(1, 0)
'''

#                         3rd way

#  get nth fibonacci number

def fibonacci(n):
    if n<=1:
        return n
    else:
        return fibonacci(n-1) + fibonacci(n-2)
print(fibonacci(19))